lib/Modules/ADCharmUtils/ADCharmUtils.psm1

```diff
@@ -101,33 +96,33 @@ function Get-MyADCredentials {
 }
 
 function Get-ActiveDirectoryContext {
-    $blobKey = ("djoin-" + $COMPUTERNAME)
+    if($global:JUJU_AD_RELATION_CONTEXT) {
+        return $global:JUJU_AD_RELATION_CONTEXT
+    }
+    $blobKey = ("djoin-" + [System.Net.Dns]::GetHostName())
     $requiredCtx = @{
-        "already-joined-$COMPUTERNAME" = $null
+        "already-joined-${env:COMPUTERNAME}" = $null
```

Comment:
In the most recent commits, this got updated and we are using only `[System.Net.Dns]::GetHostName()` calls since it gives the correct FQDN of the computer. `$env:COMPUTERNAME` gives netbios name and it was giving troubles especially when joining an AD domain (for example: we ended up with a having a mix of lowercase and uppercase hostnames).